commands/core/twig.drush.inc

```diff
@@ -0,0 +1,191 @@
+    ),
+    'core' => array('8+'),
+  );
+  return $items;
+}
+
+function drush_twig_compile() {
+  $bootstrap = \Drush::bootstrap();
+  if (!drush_get_option('standalone')) {
+    drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_FULL);
+    $searchpaths = $bootstrap->commandfile_searchpaths(DRUSH_BOOTSTRAP_DRUPAL_FULL);
+
+  }
+  else {
+    // Scan all modules and themes. By (ab)using DRUSH_BOOTSTRAP_DRUPAL_SITE we
+    // lose the ability to exclude modules, but no longer require database access
+    // to get the info of active themes. Since compiling templates is really
+    // cheap this is a valuable tradeoff.
+    // With this method we need to make sure to search core directories too.
+    $searchpaths = array(
+      'core/themes',
+      'core/modules',
+      'core/profiles',
+      'profiles',
+    );
+    $searchpaths += $bootstrap->commandfile_searchpaths(DRUSH_BOOTSTRAP_DRUPAL_SITE);
+    // We have this command set to use DRUSH_BOOTSTRAP_DRUPAL SITE. This let's us
+    // run the command and intervene before further bootstrapping into the
+    // _CONFIGURATION and _FULL phases which we do here manually.
+    drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_CONFIGURATION);
+
+    // Intervene in settings.php here to skip all database stuff.
+    $settings = Settings::getAll();
+    $settings_object = Settings::getInstance();
+    $ref = new ReflectionProperty($settings_object, 'storage');
+    $ref->setAccessible(TRUE);
+    $settings = drush_twig_custom_settings($settings);
+    $ref->setValue($settings_object, $settings);
+
+    // Add a service provider which will nullify the KeyValue store for us,
+    // because that no longer has a nice convenient $settings override.
+    $GLOBALS['conf']['container_service_providers'][] = new class() implements ServiceProviderInterface, ServiceModifierInterface {
+      public function alter(ContainerBuilder $container) {
+        $container->setParameter('factory.keyvalue', array('default' => 'keyvalue.expirable.null'));
+        $container->setParameter('factory.keyvalue.expirable', array('default' => 'keyvalue.expirable.null'));
+
+        $container->register('config.storage', 'Drupal\Core\Config\NullStorage');
+
+        $container->removeDefinition('database');
+        $container->register('database')->setSynthetic(TRUE);
+      }
+
+      public function register(ContainerBuilder $container) {
+        $container->register('keyvalue.expirable.null', 'Drupal\Core\KeyValueStore\KeyValueNullExpirableFactory');
+      }
+    };
+
+    // Skip the database bootstrapping phase.
+    drush_set_context('DRUSH_BOOTSTRAP_VALIDATION_PHASE', DRUSH_BOOTSTRAP_DRUPAL_DATABASE);
+    drush_set_context('DRUSH_BOOTSTRAP_PHASE', DRUSH_BOOTSTRAP_DRUPAL_DATABASE);
+    // Pretend to have a working site.
+    drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_FULL);
```

Comment:
How about:
```
$bootstrap->bootstrap_drupal_full();
```
You can keep `drush_bootstrap(DRUSH_BOOTSTRAP_DRUPAL_CONFIGURATION);`, but if you just call the `$bootstrap` method directly, then you shouldn't need to call `drush_set_context` to perturb Drush's bootstrapping behavior. However, I wonder what you are really skipping here, as the Drush `database` bootstrap phase [does not do much in Drupal 8](https://github.com/drush-ops/drush/blob/3b1eaaa3972e2e282a61f0f68772906dc23bce37/lib/Drush/Boot/DrupalBoot8.php#L116).  Are you just skipping [database validation](https://github.com/drush-ops/drush/blob/3b1eaaa3972e2e282a61f0f68772906dc23bce37/lib/Drush/Boot/DrupalBoot.php#L475), so that your settings.php file does not need to be configured?